wsgi_server.py

```diff
@@ -0,0 +1,96 @@
+import os
+import socket
+from enum import Enum
+from socket import SOL_SOCKET, SO_REUSEADDR
+from threading import Thread
+from typing import Callable, List, Tuple
+
+from request import HttpRequest
+from response import HttpResponse, HttpResponse404
+from wsgi_application import WSGIApplication
+
+DOCUMENT_ROOT = "./resource/server"
+
+
+class ServerMessage(Enum):
+    WAITING_CONNECTION = "Waiting for connection from client."
+    CONNECTED = "Client connected."
+    CONNECTION_CLOSED = "Connection closed."
+
+
+class WSGIServer:
+    def __init__(self, application):
+        self.env: dict = {}
```

Comment:
インスタンス変数のtype annotationは、クラス直下に
```suggestion
class WSGIServer:
	env: dict
    def __init__(self, application):
        self.env = {}
```
と書くのが正しいです。

今の書き方だと、env以外の値も警告なしに再代入できちゃうと思います

例えば、
```suggestion
class WSGIServer:
    def __init__(self, application):
        self.env: dict = {}
        self.env = "hoge"
```
としてみてください。